tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraLongLivedTokenAuthenticationContract.scala

```diff
@@ -0,0 +1,99 @@
+package com.linagora.tmail.james.common
+
+import com.linagora.tmail.james.common.LinagoraLongLivedTokenAuthenticationContract.{BASE_PATH, DEVICE_ID}
+import io.netty.handler.codec.http.HttpHeaderNames.ACCEPT
+import io.restassured.RestAssured.{`given`, requestSpecification}
+import io.restassured.authentication.NoAuthScheme
+import io.restassured.filter.log.LogDetail
+import io.restassured.http.ContentType.JSON
+import io.restassured.http.Header
+import net.javacrumbs.jsonunit.assertj.JsonAssertions.assertThatJson
+import org.apache.http.HttpStatus
+import org.apache.http.HttpStatus.SC_OK
+import org.apache.james.GuiceJamesServer
+import org.apache.james.jmap.rfc8621.contract.Fixture.{ACCEPT_RFC8621_VERSION_HEADER, ACCOUNT_ID, AUTHORIZATION_HEADER, BOB, BOB_PASSWORD, DOMAIN, baseRequestSpecBuilder}
+import org.apache.james.utils.DataProbeImpl
+import org.junit.jupiter.api.{BeforeEach, Test}
+
+object LinagoraLongLivedTokenAuthenticationContract {
+  val BASE_PATH: String = "/token"
+  val DEVICE_ID: String = "samsung1"
+}
+
+trait LinagoraLongLivedTokenAuthenticationContract {
+
+  @BeforeEach
+  def setUp(server: GuiceJamesServer): Unit = {
+    server.getProbe(classOf[DataProbeImpl])
+      .fluent()
+      .addDomain(DOMAIN.asString())
+      .addUser(BOB.asString(), BOB_PASSWORD)
+
+    requestSpecification = baseRequestSpecBuilder(server)
+      .setAuth(new NoAuthScheme)
+      //      .setBasePath(BASE_PATH)
+      .log(LogDetail.ALL)
+      .build
+  }
+
+  private def createLongLivedToken(server: GuiceJamesServer): String = {
+    val request: String =
+      s"""{
+         |  "using": ["urn:ietf:params:jmap:core", "com:linagora:params:long:lived:token"],
+         |  "methodCalls": [
+         |    ["LongLivedToken/set", {
+         |      "accountId": "$ACCOUNT_ID",
+         |      "create": {
+         |        "K38": {
+         |          "deviceId": "$DEVICE_ID"
+         |        }
+         |      }
+         |    }, "c1"]
+         |  ]
+         |}""".stripMargin
+
+    `given`()
+      .auth()
+      .basic(BOB.asString(), BOB_PASSWORD)
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .body(request)
+      .when()
+      .post()
+      .`then`
+      .statusCode(HttpStatus.SC_OK)
+      .extract()
+      .jsonPath()
+      .get("methodResponses[0][1].created.K38.token")
+  }
+
+  private def buildBearTokenHeader(longLivedToken: String): Header =
+    new Header(AUTHORIZATION_HEADER, "Bearer " + BOB.asString() + "_" + longLivedToken)
```

Comment:
IMO this is a mis-design: we should not expect the client to know he should concatenate the two strings.

Please have `LongLivedToken/set` that can be directly passed in a Bearer header

Reply:
Resolved at https://github.com/linagora/tmail-backend/pull/236